internal/isoeditor/rhcos_test.go

```diff
@@ -0,0 +1,116 @@
+package isoeditor
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	defaultTestOpenShiftVersion = "4.6"
+	defaultTestServiceBaseURL   = "http://10.35.59.36:30485"
+)
+
+func TestIsoEditor(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "IsoEditor")
+}
+
+var _ = Context("with test files", func() {
+	var (
+		isoDir   string
+		isoFile  string
+		filesDir string
+		volumeID = "Assisted123"
+		log      logrus.FieldLogger
+	)
+	BeforeSuite(func() {
+		filesDir, isoDir, isoFile = createIsoViaGenisoimage(volumeID)
+		log = getTestLog()
+	})
+
+	AfterSuite(func() {
+		os.RemoveAll(filesDir)
+		os.RemoveAll(isoDir)
+	})
+
+	Describe("CreateMinimalISOTemplate", func() {
+		It("iso created successfully", func() {
+			editor := CreateEditor(isoFile, defaultTestOpenShiftVersion, log)
+			_, err := editor.CreateMinimalISOTemplate(defaultTestServiceBaseURL)
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("missing iso file", func() {
+			editor := CreateEditor("invalid", defaultTestOpenShiftVersion, log)
+			_, err := editor.CreateMinimalISOTemplate(defaultTestServiceBaseURL)
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	Describe("AddRootFSURL", func() {
+		It("rootfs_url added successfully", func() {
+			editor := CreateEditor(isoFile, defaultTestOpenShiftVersion, log)
+			rootfsURL := fmt.Sprintf("%s/api/assisted-install/v1/boot-files?file_type=rootfs.img&openshift_version=%s", defaultTestServiceBaseURL, defaultTestOpenShiftVersion)
+			err := editor.GetIsoHandler().Extract()
```

Comment:
Instead of exposing the iso handler through this method, can we either access it through the struct (as we're in the same package) or create the iso handler explicitly in the test and use `NewEditor`?

Don't want to pollute the public API just for tests.

Reply:
sure, done